config/v1/json.go

```diff
@@ -0,0 +1,42 @@
+package v1
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"regexp"
+)
+
+// reUUID4 uses a regexp to avoid taking a dependency on a UUID library for now
+var reUUID4 = regexp.MustCompile(`^[a-fA-F\d]{8}-[a-fA-F\d]{4}-4[a-fA-F\d]{3}-[89abAB][a-fA-F\d]{3}-[a-fA-F\d]{12}$`)
+
+// UnmarshalJSON unmarshals RFC4122 uuid from string.
+func (cid *ClusterID) UnmarshalJSON(b []byte) error {
```

Comment:
are you *sure* you want to do this?  it will prevent deserialization client-side.  I would stay away from this and enforce it as a CRD validator.

Reply:
This was being enforced in CVO right now.  We won't be able to enforce this with a naive validator.

Reply:
I agree with the point